timon/state.py

```diff
@@ -202,6 +202,55 @@ def save(self, safe=True):
         if fname:
             os.rename(partial_fname, fname)
 
+    def has_state_changed(self, probe, status, flap_detection=True,
+                          flap_cnt=2):
+        """
+        Check if the probe state has changed between the current and the
+        previous run.
+        If flap detection is enabled, ensure there's <flap_cnt> successive
+        similar results before considering the state has changed.
+
+        Args:
+            probe (Probe object): Probe to check
+            status (str): state of the current run
+            flap_detection (bool, optional): activate/deactivate flap
+                detection. Defaults to True.
+            flap_cnt (int, optional): Minimum successive similar results
+                before considering state has changed (current state is counted in).
+                Defaults to 2.
+
+        Returns:
+            bool: returns True if state has changed, False otherwise
+        """
+        probe_name = probe.name
+        prb_states = self.state['probe_state']
+        if probe_name not in prb_states:
+            prb_states[probe_name] = []
+        pst = prb_states[probe_name]
+        if not pst:
+            return True
+        if not flap_detection:
+            # Compare only with the previous status
+            prev_status = pst[-1][1]
+            return status != prev_status
+        else:
+            # FLAP DETECTION
+            if len(pst) < flap_cnt:
+                logger.warning(
+                    "Cannot flap detect the status changing of the probe %s,"
```

Comment:
shouldn't this be info instead of warning.

This will systematically happen when starting a new project, so perhaps not worth a warning

Reply:
Yeah, you're right ! Changed